test/github-issues/660/issue-660.ts

```diff
@@ -0,0 +1,116 @@
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+        schemaCreate: true,
+        dropSchemaOnConnection: true,
+    }));
+    beforeEach(() => reloadTestingDatabases(connections));
+    after(() => closeTestingConnections(connections));
+
+    const user = new User();
+    user.name = "Tim Merrison";
+
+    it("should create an INSERT statement, including RETURNING clause as appropriate", () => Promise.all(connections.map(async connection => {
+
+        const sql = connection.createQueryBuilder()
+            .insert()
+            .into(User)
+            .values(user)
+            .returning("INSERTED.*")
+            .disableEscaping()
+            .getSql();
+
+        if (connection.driver instanceof SqlServerDriver) {
+            expect(sql).to.equal("INSERT INTO user(name) OUTPUT INSERTED.* VALUES (@0)"); }
+        else if (connection.driver instanceof PostgresDriver) {
+            expect(sql).to.equal("INSERT INTO user(name) VALUES ($1) RETURNING INSERTED.*"); }
+        // else { // this is arguably an error case, since .returning() is only supported by PostgreSQL and MSSQL
+        //     expect(sql).to.equal("INSERT INTO user(name) VALUES (@0)"); }
+    })));
+
+    it("should create an INSERT statement, including OUTPUT clause as appropriate", () => Promise.all(connections.map(async connection => {
+
+        const sql = connection.createQueryBuilder()
+            .insert()
+            .into(User)
+            .values(user)
+            .output("INSERTED.*")
+            .disableEscaping()
+            .getSql();
+
+        if (connection.driver instanceof SqlServerDriver) {
+            expect(sql).to.equal("INSERT INTO user(name) OUTPUT INSERTED.* VALUES (@0)"); }
+        else if (connection.driver instanceof PostgresDriver) {
+            expect(sql).to.equal("INSERT INTO user(name) VALUES ($1) RETURNING INSERTED.*"); }
+        // else { // this is arguably an error case, since .returning() is only supported by PostgreSQL and MSSQL
+        //     expect(sql).to.equal("INSERT INTO user(name) VALUES (@0)"); }
+    })));
+
+    it("should create an UPDATE statement, including RETURNING or OUTPUT clause as appropriate", () => Promise.all(connections.map(async connection => {
+
+        const sql = connection.createQueryBuilder()
+            .update(User)
+            .set({ name: "Joe Bloggs" })
+            .where("name = :name", { name: user.name })
+            .returning("INSERTED.*")
+            .disableEscaping()
+            .getSql();
+
+        if (connection.driver instanceof SqlServerDriver) {
+            expect(sql).to.equal("UPDATE user SET name = @0 OUTPUT INSERTED.* WHERE name = @1"); }
+        else if (connection.driver instanceof PostgresDriver) {
+            expect(sql).to.equal("UPDATE user SET name = $1 RETURNING INSERTED.* WHERE name = $2"); } // Fudged, until PostgreSQL RETURNING clause can be correctly placed after WHERE clause
```

Comment:
I am now able to run the PostgreSQL test locally, and the INSERT seems to work correctly, but UPDATE does not, because, unlike for SQLServer, PostgreSQL wants the RETURNING clause at the end (e.g. after any WHERE clause), and I'm not sure how to achieve that, and our advice would be greatly appreciated!